client/src/Movies/Movie.js

```diff
@@ -25,17 +25,21 @@ export default class Movie extends Component {
         console.error(error);
       });
   };
-  // Uncomment this code when you're ready for the stretch problems
-  // componentWillReceiveProps(newProps){
-  //   if(this.props.match.params.id !== newProps.match.params.id){
-  //     this.fetchMovie(newProps.match.params.id);
-  //   }
-  // }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
+  componentDidMount() {
+    this.fetchMovie(this.props.match.params.id)
+  } 
+  
+  componentWillReceiveProps(newProps){
+    if(this.props.match.params.id !== newProps.match.params.id){
+      this.fetchMovie(newProps.match.params.id);
+    }
+  }
```

Comment:
If you have time, please check on this life cycle method on React documentation. This is a life cycle on the old version of React and they're moving away from it. Note: I know this is part of the given code, just want you to check on what it does.